docs/cip/CIP-8_Matrix_Packages.md

```diff
@@ -0,0 +1,209 @@
+
+TODO i am totally out of my depth with quantization and need to run this through with someone. Key questions - does it make sense to try to reduce embedding and query pipeline fns to a model + some set of constraint arguments? Is quantization distinct from embedding - i.e. do we have one column for each in the parquet file?
+TODO finish
+
+#### `data.digests`
+```yaml
+  digests:
+    data: "sha256:decafbad"
+```
+
+### `matrix.data.parquet`
+
+The `matrix.data.parquet` file contains the core dataset of the matrix package. Each row in this Parquet file closely mirrors Chroma database content. The following columns are expected:
+
+- `src`- **Original Document Source**: This field holds the original source data, which could be text, documents, or any other relevant information that forms the basis of the dataset. This maps directly to a Chroma document.
+- `embedding`- **Computed Embeddings**: Computed embeddings are generated from the original document source using the model specified in `matrix.package.yaml`. This maps directly to Chroma embeddings.
+- `meta`- **Metadata**: Metadata associated with each row is a set of key/value pairs that provide context or information about the data. This metadata could include timestamps, data source references, or any other relevant details. This maps directly to Chroma metadata.
+
+For now, the `matrix.data.parquet` file is required to physically exist adjacent to the `matrix.package.yaml` file. This is an initial implementation simplification and convenience. It is expected that a follow-up shortly after this CIP is implemented will add remote storage support, facilitated by the use of OCI registries.
+
+In this scenario, the actual `matrix.data.parquet` file would be stored remotely, and the matrix package would include references or pointers to the remote location rather than including the dataset itself. This approach would reduce package size and enhance efficiency in distributing and managing large datasets, making the data file optional in such cases.
+
+TODO address degrees of freedom with parquet - file metadata, multi-file splitting, etc.
+### Chroma Client & CLI Changes
+
+For matrix packages to be useful, the procedure for producing and consuming them must be trivial. This section describes changes to both the Chroma client and CLI in order to support matrix packages.
+
+TODO lollll so much to fill in - sdboyer can do some, but also others/jeffchuber?
+#### User story
+
+TODO describe basic user story/interaction loop, from initial data loading into chroma, massaging, then `chroma export-matrix` , then someone else doing a `chroma import-matrix`, then running their app
+
+#### Import and `chroma import-matrix`
+
+Idempotence is probably critical here
+
+TODO do this
+
+#### Export and `chroma export-matrix`
+
+TODO do this
+
+## Alternatives Considered
```

Comment:
What are the downsides of using standard HF-style datasets? E.g. https://huggingface.co/datasets/KShivendu/dbpedia-entities-openai-1M
These seem to be pretty widely used and if we can achieve inter-op the whole community (not just Chroma) can benefit from these datasets.

Reply:
I think the primary difference is storage of the index and other metadata.

Reply:
Ah! I ran across HF datasets a bit, but didn't dig on them at all. This is definitely one of those "sam is a total newbie to 80% of what's happening here" things.

Looking quickly over the HF dataset docs, it does seem like this is covering a _lot_ of what i was wishing just existed as some standard we could build atop. Which goes to @HammadB's point - there is definitely something additional we're talking about with matrix packages.

So, feels to me like the research question here is how cleanly we could layer our needs on top. (In particular i have this "sparse, distributed cache" possible-goal in mind, which needs fleshing out...somewhere)